src/AlertsManagement/AlertsManagement/AlertsManagementBaseCmdlet.cs

```diff
@@ -0,0 +1,183 @@
+        /// <param name="withTimeStamp">true if the message should include a timestamp, false (default) it no timestamp should be included</param>
+        protected void WriteIdentifiedWarning(string cmdletName, string topic, string message, bool withTimeStamp = false)
+        {
+            string formattedMessage = string.Format(
+                CultureInfo.InvariantCulture,
+                "[{0}] {1}: {2}",
+                cmdletName,
+                topic,
+                message);
+
+            if (withTimeStamp)
+            {
+                WriteWarningWithTimestamp(formattedMessage);
+            }
+            else
+            {
+                WriteWarning(formattedMessage);
+            }
+        }
+
+        /// <summary>
+        /// Execute the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            string code = null;
+            HttpStatusCode? statusCode = null;
+            string reasonPhrase = null;
+            string message = null;
+            string exName = null;
+
+            try
+            {
+                this.ProcessRecordInternal();
+            }
+            catch (AggregateException ex)
+            {
+                // Process the exception to be as informative as possible to the user
+                var exTemp = ex.Flatten().InnerException ?? ex;
+                exName = exTemp.GetType().Name;
+                message = exTemp.Message;
+
+                if (exTemp is RestException)
+                {
+                    // All the following Exceptions have the same structure, but their common ancestor does not contain Body nor Response
+                    var cloudException = exTemp as CloudException;
+                    if (cloudException != null)
+                    {
+                        message = cloudException.Body.Message;
+                        code = cloudException.Body.Code;
+                        statusCode = cloudException.Response.StatusCode;
+                        reasonPhrase = cloudException.Response.ReasonPhrase;
+                    }
+                    else
+                    {
+                        // TODO: Enable this flow once SDK is published
+                        
+                        // New model to report errors (from Swagger Spec)
+                        var errorResponse = exTemp as Microsoft.Azure.Management.AlertsManagement.Models.ErrorResponseException;
+                        if (errorResponse != null)
+                        {
+                            statusCode = errorResponse.Response.StatusCode;
+                            reasonPhrase = errorResponse.Response.ReasonPhrase;
+                        }
+                        else
+                        {
+                            message = exTemp.ToString();
+                        }
+                    }
+                }
+
+                throw new PSInvalidOperationException(
```

Comment:
You should not wrap cloud exceptiosn this way.  Cloud exceptions should be thrown directly, there is a cmdlet that will unwrap these and extract data the user can emply for debugging the issue

Reply:
Makes sense. Committed